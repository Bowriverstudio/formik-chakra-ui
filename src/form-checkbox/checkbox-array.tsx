import { Checkbox, CheckboxProps } from '@chakra-ui/core';
import { useField } from 'formik';
import React, { FC } from 'react';
import { BaseProps } from '../base-props';

export type FormCheckboxArrayProps = BaseProps & CheckboxProps;

export const FormCheckboxArray: FC<FormCheckboxArrayProps> = (
  props: FormCheckboxArrayProps
) => {
  const { name, label, my, children, ...rest } = props;
  const [field, { error, touched }] = useField(name);
  const isChecked = field.value?.includes(props.value) ?? false;

  return (
    <Checkbox
      id={name}
      isInvalid={!!error && touched}
      isChecked={isChecked}
      {...field}
      {...rest}
    >
      {children}
    </Checkbox>
  );
};

export default FormCheckboxArray;
