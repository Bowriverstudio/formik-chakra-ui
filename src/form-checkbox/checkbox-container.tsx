import { CheckboxProps, Stack, StackProps } from '@chakra-ui/core';
import React, { FC, ReactNode } from 'react';
import { BaseProps } from '../base-props';
import { FormControl } from '../form-control';

export type CheckboxContainerProps = BaseProps &
  CheckboxProps & { stackConfig?: StackProps; children: ReactNode };

export const CheckboxContainer: FC<CheckboxContainerProps> = (
  props: CheckboxContainerProps
) => {
  const { name, label, my, children, stackConfig, ...rest } = props;

  return (
    <FormControl name={name} label={label} my={my} {...rest}>
      <Stack pl={6} mt={1} spacing={1} {...stackConfig}>
        {children}
      </Stack>
    </FormControl>
  );
};

export default CheckboxContainer;
