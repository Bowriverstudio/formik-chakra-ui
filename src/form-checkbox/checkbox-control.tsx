import { Checkbox, CheckboxProps } from '@chakra-ui/core';
import { useField } from 'formik';
import React, { FC } from 'react';
import { BaseProps } from '../base-props';

type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;

export type CheckboxArrayProps = BaseProps &
  Overwrite<CheckboxProps, { value: string | number }>;

export const CheckboxControl: FC<CheckboxArrayProps> = (
  props: CheckboxArrayProps
) => {
  const { name, label, my, children, ...rest } = props;
  const [field, { error, touched }] = useField(name);
  let isChecked;
  if (field.value instanceof Array) {
    isChecked = field.value.includes(props.value) ?? false;
  }

  return (
    <Checkbox
      id={name}
      isInvalid={!!error && touched}
      isChecked={isChecked}
      {...field}
      {...rest}
    >
      {label}
      {children}
    </Checkbox>
  );
};

export default CheckboxControl;
