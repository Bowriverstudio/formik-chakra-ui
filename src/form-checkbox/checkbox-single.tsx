import { Checkbox, CheckboxProps } from '@chakra-ui/core';
import { useField } from 'formik';
import React, { FC, ReactNode } from 'react';
import { BaseProps } from '../base-props';
import { FormControl } from '../form-control';

export type CheckboxSingleProps = BaseProps & {
  checkBoxProps?: CheckboxProps;
  children: ReactNode;
};

export const CheckboxSingleControl: FC<CheckboxSingleProps> = (
  props: CheckboxSingleProps
) => {
  const { name, label, children, formControlProps, checkBoxProps } = props;
  const [field, { error, touched }] = useField(name);
  const isChecked = field.value;

  return (
    <FormControl name={name} {...formControlProps}>
      <Checkbox
        id={name}
        isInvalid={!!error && touched}
        isChecked={isChecked}
        {...field}
        {...checkBoxProps}
      >
        {label}
        {children}
      </Checkbox>
    </FormControl>
  );
};
