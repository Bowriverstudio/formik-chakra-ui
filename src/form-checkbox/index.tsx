import { Checkbox, CheckboxProps } from '@chakra-ui/core';
import { useField } from 'formik';
import React, { FC } from 'react';
import { BaseProps } from '../base-props';
import { FormControl } from '../form-control';

export type FormCheckboxProps = BaseProps & CheckboxProps;

export const FormCheckbox: FC<FormCheckboxProps> = (
  props: FormCheckboxProps
) => {
  const { name, label, my, ...rest } = props;
  const [field] = useField(name);
  const isChecked = field.value;

  return (
    <FormControl name={name} my={my}>
      <Checkbox id={name} isChecked={isChecked} {...field} {...rest}>
        {label}
      </Checkbox>
    </FormControl>
  );
};

export * from './checkbox-container';
export * from './checkbox-array';
export default FormCheckbox;
