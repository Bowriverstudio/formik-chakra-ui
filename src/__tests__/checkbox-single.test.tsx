import { CSSReset, ThemeProvider } from '@chakra-ui/core';
import { render } from '@testing-library/react';
import { Form, Formik } from 'formik';
import * as React from 'react';
import { CheckboxSingleControl } from '../form-checkbox';

test('CheckboxSingleControl Renders Correctly', () => {
  const { asFragment } = render(
    <Formik onSubmit={() => {}} initialValues={{ test: false }}>
      <Form>
        <ThemeProvider>
          <CSSReset />
          <CheckboxSingleControl name="test">Test Label</CheckboxSingleControl>
        </ThemeProvider>
      </Form>
    </Formik>
  );

  expect(asFragment()).toMatchSnapshot();
});
