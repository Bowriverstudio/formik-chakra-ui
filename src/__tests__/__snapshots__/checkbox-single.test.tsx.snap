// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxSingleControl Renders Correctly 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="css-17t5ffy"
      role="group"
    >
      <label
        class="css-jwmyb8"
      >
        <input
          aria-checked="false"
          class="css-1gy22ay"
          id="test"
          name="test"
          type="checkbox"
          value="false"
        />
        <div
          aria-hidden="true"
          class="css-nlndso"
        >
          <svg
            class="css-1vhqk79"
            focusable="false"
            role="presentation"
            viewBox="0 0 14 14"
          >
            <g
              fill="currentColor"
            >
              <polygon
                points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
              />
            </g>
          </svg>
        </div>
        <div
          class="css-6qsuox"
        >
          Test Label
        </div>
      </label>
    </div>
  </form>
</DocumentFragment>
`;
